engine/crates/graphql-extensions/src/runtime_log.rs

```diff
@@ -97,23 +97,19 @@ impl Extension for RuntimeLogExtension {
             _ => None,
         });
 
-        self.log_event_receiver
-            .invoke(
-                runtime_ctx.ray_id(),
-                runtime_ctx.log.request_log_event_id,
-                LogEventType::OperationCompleted {
-                    name: operation_name.map(From::from),
-                    duration,
-                    r#type: match response.operation_type {
-                        ParserOperationType::Query => OperationType::Query {
-                            is_introspection: crate::is_operation_introspection(operation),
-                        },
-                        ParserOperationType::Mutation => OperationType::Mutation,
-                        ParserOperationType::Subscription => OperationType::Subscription,
+        if let Some(operation) = response.graphql_operation.as_ref() {
+            self.log_event_receiver
+                .invoke(
+                    runtime_ctx.ray_id(),
+                    runtime_ctx.log.request_log_event_id,
+                    LogEventType::OperationCompleted {
+                        name: operation_name.map(From::from),
+                        duration,
+                        r#type: operation.r#type,
```

Comment:
Am I right to think this simplification is the motivation for the change?

Reply:
This, plus getting rid of [this API-side processing](https://github.com/grafbase/api/pull/2852/files#diff-666c2f5e9ac8daf19cde946d15b8f52b36192dbdef530ef7fab2eed76571a8f4L823-L870) here.